### PR TITLE
Feature/pet functions refactor

### PR DESCRIPTION
**Pull request by marclll**

### CODE REVIEW

File: src/test/java/org/pesmypetcare/webservice/builders/PathTest.java [hunk 1]
```diff
@@ -82,13 +82,15 @@ public void buildUsersCollectionPath() {
...
-        assertEquals("groups/" + groupId + "/forums/" + forumId, path, "Should return the path to the forum.");
+        assertEquals("groups/" + groupId + "/forums/" + forumId, path
+            , "Should return the path to the forum.");
     }
 
     @Test
     public void buildForumsCollectionPath() {
         String path = Path.ofCollection(Collections.forums, groupId);
-        assertEquals("groups/" + groupId + "/forums", path, "Should return the path to a forums collection.");
+        assertEquals("groups/" + groupId + "/forums", path
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/builders/PathTest.java:92]: Codacy Issue found: [The String "/forums" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 12
           

Coverage increased per file
===========================
+ src/main/java/org/pesmypetcare/webservice/builders/CollectionPath.java  2
         

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/builders/Path.java  4
- src/main/java/org/pesmypetcare/webservice/builders/CollectionPath.java  4
- src/main/java/org/pesmypetcare/webservice/builders/DocumentPath.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446&bid=17965954)

File: src/test/java/org/pesmypetcare/webservice/builders/PathTest.java [hunk 2]
```diff
@@ -82,13 +82,15 @@ public void buildUsersCollectionPath() {
     @Test
     public void buildForumPath() {
         String path = Path.ofDocument(Collections.forums, groupId, forumId);
-        assertEquals("groups/" + groupId + "/forums/" + forumId, path, "Should return the path to the forum.");
+        assertEquals("groups/" + groupId + "/forums/" + forumId, path
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/builders/PathTest.java:85]: Codacy Issue found: [The String "/forums/" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446)

File: src/test/java/org/pesmypetcare/webservice/builders/PathTest.java [hunk 3]
```diff
@@ -102,32 +104,36 @@ public void buildForumNamePath() {
     @Test
     public void buildForumNamesCollectionPath() {
         String path = Path.ofCollection(Collections.forumsNames, groupName);
-        assertEquals("groups_names/" + groupName + "/forums", path,
-            "Should return the path to the collectionof forum names.");
+        assertEquals("groups_names/" + groupName + "/forums", path
+            , "Should return the path to the collectionof forum names.");
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/builders/PathTest.java:108]: Codacy Issue found: [',' is preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446)

File: src/test/java/org/pesmypetcare/webservice/builders/PathTest.java [hunk 4]
```diff
@@ -102,32 +104,36 @@ public void buildForumNamePath() {
...
     }
 
     @Test
     public void buildPetPath() {
         String path = Path.ofDocument(Collections.pets, userId, petName);
-        assertEquals("users/" + userId + "/pets/" + petName, path, "Should return the path to the pet.");
+        assertEquals("users/" + userId + "/pets/" + petName, path
+            , "Should return the path to the pet.");
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/builders/PathTest.java:115]: Codacy Issue found: [',' is preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446)

File: src/test/java/org/pesmypetcare/webservice/builders/PathTest.java [hunk 5]
```diff
@@ -102,32 +104,36 @@ public void buildForumNamePath() {
...
+        assertEquals("users/" + userId + "/pets", path
+            , "Should return the path to the pets collection.");
     }
 
     @Test
     public void buildMemberPath() {
         String path = Path.ofDocument(Collections.members, groupId, userId);
-        assertEquals("groups/" + groupId + "/members/" + userId, path, "Should return the path to the member.");
+        assertEquals("groups/" + groupId + "/members/" + userId, path
+            , "Should return the path to the member.");
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/builders/PathTest.java:129]: Codacy Issue found: [',' is preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446)

File: src/test/java/org/pesmypetcare/webservice/builders/PathTest.java [hunk 6]
```diff
@@ -203,9 +209,9 @@ public void buildFrequencyOfWashesEntriesCollectionPath() {
     @Test
     public void buildMedicationEntryPath() {
         String medicationName = "painkillers";
-        String path = Path.ofDocument(Collections.medications, userId, petName, date, medicationName);
-        assertEquals("users/" + userId + "/pets/" + petName + "/medications/" + date + "½" + medicationName, path,
-            "Should return the path to the medication entry.");
+        String path = Path.ofDocument(Collections.medications, userId, petName, date + "-" + medicationName);
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/builders/PathTest.java:212]: Codacy Issue found: [The String "-" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446)

File: src/test/java/org/pesmypetcare/webservice/builders/PathTest.java [hunk 7]
```diff
@@ -82,13 +82,15 @@ public void buildUsersCollectionPath() {
     @Test
     public void buildForumPath() {
         String path = Path.ofDocument(Collections.forums, groupId, forumId);
-        assertEquals("groups/" + groupId + "/forums/" + forumId, path, "Should return the path to the forum.");
+        assertEquals("groups/" + groupId + "/forums/" + forumId, path
+            , "Should return the path to the forum.");
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/builders/PathTest.java:86]: Codacy Issue found: [',' is preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446)

File: src/test/java/org/pesmypetcare/webservice/builders/PathTest.java [hunk 8]
```diff
@@ -102,32 +104,36 @@ public void buildForumNamePath() {
...
+        assertEquals("users/" + userId + "/pets/" + petName, path
+            , "Should return the path to the pet.");
     }
 
     @Test
     public void buildPetsCollectionPath() {
         String path = Path.ofCollection(Collections.pets, userId);
-        assertEquals("users/" + userId + "/pets", path, "Should return the path to the pets collection.");
+        assertEquals("users/" + userId + "/pets", path
+            , "Should return the path to the pets collection.");
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/builders/PathTest.java:122]: Codacy Issue found: [',' is preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446)

File: src/test/java/org/pesmypetcare/webservice/builders/PathTest.java [hunk 9]
```diff
@@ -102,32 +104,36 @@ public void buildForumNamePath() {
...
+        assertEquals("groups/" + groupId + "/members/" + userId, path
+            , "Should return the path to the member.");
     }
 
     @Test
     public void buildMembersCollectionPath() {
         String path = Path.ofCollection(Collections.members, groupId);
-        assertEquals("groups/" + groupId + "/members", path, "Should return the path to the members collection.");
+        assertEquals("groups/" + groupId + "/members", path
+            , "Should return the path to the members collection.");
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/builders/PathTest.java:136]: Codacy Issue found: [',' is preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446)

File: src/main/java/org/pesmypetcare/webservice/builders/DocumentPath.java [hunk 10]
```diff
@@ -256,60 +257,18 @@ private static StringBuilder buildPathToFreqTraining(@NonNull String userId, @No
...
+     * @param key Key to the collection entry
      * @return The path
      */
     @NonNull
-    private static StringBuilder buildPathToMeal(@NonNull String userId, @NonNull String petName,
-                                                 @NonNull String date) {
-        return buildPathToPet(userId, petName).append("/meals/").append(date);
+    private static StringBuilder buildPathToPetCollection(@NonNull String userId, @NonNull String petName,
+                                                          @NonNull String collectionName, @NonNull String key) {
+        return buildPathToPet(userId, petName).append("/").append(collectionName).append("/").append(key);
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/builders/DocumentPath.java:271]: Codacy Issue found: [The String "/" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446)

File: src/test/java/org/pesmypetcare/webservice/builders/PathTest.java [hunk 11]
```diff
@@ -82,13 +82,15 @@ public void buildUsersCollectionPath() {
...
+        assertEquals("groups/" + groupId + "/forums/" + forumId, path
+            , "Should return the path to the forum.");
     }
 
     @Test
     public void buildForumsCollectionPath() {
         String path = Path.ofCollection(Collections.forums, groupId);
-        assertEquals("groups/" + groupId + "/forums", path, "Should return the path to a forums collection.");
+        assertEquals("groups/" + groupId + "/forums", path
+            , "Should return the path to a forums collection.");
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/builders/PathTest.java:93]: Codacy Issue found: [',' is preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446)

File: src/test/java/org/pesmypetcare/webservice/builders/PathTest.java [hunk 12]
```diff
@@ -102,32 +104,36 @@ public void buildForumNamePath() {
...
-            "Should return the path to the collectionof forum names.");
+        assertEquals("groups_names/" + groupName + "/forums", path
+            , "Should return the path to the collectionof forum names.");
     }
 
     @Test
     public void buildPetPath() {
         String path = Path.ofDocument(Collections.pets, userId, petName);
-        assertEquals("users/" + userId + "/pets/" + petName, path, "Should return the path to the pet.");
+        assertEquals("users/" + userId + "/pets/" + petName, path
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/builders/PathTest.java:114]: Codacy Issue found: [The String "/pets/" appears 25 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/builders/Path.java  4
- src/main/java/org/pesmypetcare/webservice/builders/DocumentPath.java  4
- src/main/java/org/pesmypetcare/webservice/builders/CollectionPath.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446&bid=17965954)

File: src/test/java/org/pesmypetcare/webservice/builders/PathTest.java [hunk 13]
```diff
@@ -82,13 +82,15 @@ public void buildUsersCollectionPath() {
...
-        assertEquals("groups/" + groupId + "/forums/" + forumId, path, "Should return the path to the forum.");
+        assertEquals("groups/" + groupId + "/forums/" + forumId, path,
+            "Should return the path to the forum.");
     }
 
     @Test
     public void buildForumsCollectionPath() {
         String path = Path.ofCollection(Collections.forums, groupId);
-        assertEquals("groups/" + groupId + "/forums", path, "Should return the path to a forums collection.");
+        assertEquals("groups/" + groupId + "/forums", path,
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/builders/PathTest.java:92]: Codacy Issue found: [The String "/forums" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446)

File: src/test/java/org/pesmypetcare/webservice/builders/PathTest.java [hunk 14]
```diff
@@ -109,25 +111,29 @@ public void buildForumNamesCollectionPath() {
     @Test
     public void buildPetPath() {
         String path = Path.ofDocument(Collections.pets, userId, petName);
-        assertEquals("users/" + userId + "/pets/" + petName, path, "Should return the path to the pet.");
+        assertEquals("users/" + userId + "/pets/" + petName, path,
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/builders/PathTest.java:114]: Codacy Issue found: [The String "/pets/" appears 25 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446)

File: src/test/java/org/pesmypetcare/webservice/builders/PathTest.java (hunk 6 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/builders/PathTest.java:212]: Codacy Issue found: [The String "-" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446)

File: src/test/java/org/pesmypetcare/webservice/builders/PathTest.java [hunk 15]
```diff
@@ -82,13 +82,15 @@ public void buildUsersCollectionPath() {
     @Test
     public void buildForumPath() {
         String path = Path.ofDocument(Collections.forums, groupId, forumId);
-        assertEquals("groups/" + groupId + "/forums/" + forumId, path, "Should return the path to the forum.");
+        assertEquals("groups/" + groupId + "/forums/" + forumId, path,
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/builders/PathTest.java:85]: Codacy Issue found: [The String "/forums/" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446)

File: src/main/java/org/pesmypetcare/webservice/builders/DocumentPath.java (hunk 10 above)
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/builders/DocumentPath.java:271]: Codacy Issue found: [The String "/" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446)

**Review by santidrj — APPROVED** [review]
:+1:

**Review by oricat8 — APPROVED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ src/main/java/org/pesmypetcare/webservice/builders/CollectionPath.java  2
         

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/builders/CollectionPath.java  4
- src/main/java/org/pesmypetcare/webservice/builders/Path.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5482446&bid=17965954)

**Pull request merged**